### ISSUE TITLE
Random failure when building in parallel

### ISSUE THREAD

**Issue by samueloph** [aircrack-ng/mdk4#38]
Forwarding bug reported in: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=921694

I didn't reproduce the problem yet but I think it's worth reporting upstream as the reporter is reliable and you may already have more information or found this problem already.

Reporter: Santiago Vila <sanvila@debian.org>
```
The failure happens randomly. Sometimes it fails, sometimes it does not.

I can reproduce the randomness on 1-S instances from Scaleway (having
two cores). I have also tried on 1-XS instances (having a single
core), but the failure never happened there so far. (This is the
reason I've put the word "parallel" in the subject, but that's just a
working hypothesis for now).

I've put a bunch of failed build logs here:

https://people.debian.org/~sanvila/build-logs/mdk4/

BTW: One of them contains this funny error message:

ranlib: out of memory allocating 42949672960 bytes after a total of 143360 bytes

That's 42 GB of memory. Could this be a bug in binutils?

If you need a test machine to reproduce this, please contact me privately,
I could provide ssh access to a machine where this randomness happens.

There is still another approach at fixing this problem, which would be
to assume it is a Makefile bug (my current bet), and then try to see
how libosdep.a could be misbuilt (maybe a race condition). You could,
for example, try to prove that the Makefile is correct, and by doing
so, realize what's wrong in it. Of course that will only work if
the failure is due to a Makefile bug.
```

**Comment by E7mer** [comment]
Thank you for reporting this problem, I didn't reproduce the problem yet, and I will test more.

**Comment by samueloph** [comment]
@E7mer I'm gonna package the latest git snapshot of mdk4 on Debian and ask the bug reporter to check if this is still happening. Hold on for updates.

**Comment by samueloph** [comment]
Closing this as there was no response from the Debian bug submitter regarding the issue happening on the latest release, I'm gonna assume that the problem is fixed in 4.1
